lib/vector-config-macros/src/configurable.rs

```diff
@@ -350,36 +360,25 @@ fn generate_variant_metadata(
     meta_ident: &Ident,
     variant: &Variant<'_>,
 ) -> proc_macro2::TokenStream {
-    let variant_as_configurable = get_variant_type_as_configurable();
-
     let maybe_title = get_metadata_title(meta_ident, variant.title());
     let description = get_metadata_description(meta_ident, variant.description())
         .expect("enum variants without a description should be rejected during AST parsing");
     let maybe_deprecated = get_metadata_deprecated(meta_ident, variant.deprecated());
     let maybe_custom_attributes = get_metadata_custom_attributes(meta_ident, variant.metadata());
 
+    // We ase a unit tuple here because we don't have access to the actual type of the variant's enum container,
+    // at least not without parsing it specifically as a type token from an amalgamated version of the enum ident... but
+    // it doesn't actually matter because enums don't support default values, which is the only spot in `Metadata<T>`
+    // where the `T` gets used, so we can carry all the other pertinent details with `()`... it's still a littlw wonky, though.
     quote! {
-        let mut #meta_ident = #variant_as_configurable::metadata();
+        let mut #meta_ident =  <() as ::vector_config::Configurable>::metadata();
         #maybe_title
         #description
         #maybe_deprecated
         #maybe_custom_attributes
     }
 }
 
-fn get_field_type_as_configurable(field: &Field<'_>) -> proc_macro2::TokenStream {
-    let field_ty = field.ty();
-    quote! { <#field_ty as ::vector_config::Configurable> }
-}
-
-fn get_variant_type_as_configurable() -> proc_macro2::TokenStream {
-    // We hardcode this to a unit tuple because we don't have access to the actual type of the variant's enum container,
-    // at least not without parsing it specifically as a type token from an amalgamated version of the enum ident... but
-    // it doesn't actually matter because enums don't support default values, which is the only spot in `Metadata<T>`
-    // where the `T` gets used, so we can carry all the other pertinent details with `()`... it's still a littlw wonky, though.
-    quote! { <() as ::vector_config::Configurable> }
-}
-
```

Comment:
I'm curious why you moved from these functions to what appears to be their original output in the final `quote!` macros.

Reply:
This got removed because I couldn't actually get it to generate the correct spanned output and had to group everything together in a single `quote_spanned!` rather than doing it piecemeal.

Essentially, I tried to do all of the `quote_spanned!` stuff in these functions first, and use them in an interpolated fashion, but it just... didn't work.